### PR TITLE
Fix: AutoAdd in Dismantle (#148)

### PR DESCRIPTION
**Pull request by natoweg**
- added check for Gacha or Locked items that should not be dismantle-able
- Fix: #148

### CODE REVIEW

**Review by AngeloTadeucci — APPROVED** [review]

**Review by Zintixx — COMMENTED** [review]
Looks good. Add a check for `enableBreak` too. If the GachadismantleId is > 0, it ignores this bool.

File: Maple2.Server.Game/PacketHandlers/ItemDismantleHandler.cs [hunk 1]
```diff
@@ -7,6 +7,7 @@
 using Maple2.Server.Core.PacketHandlers;
 using Maple2.Server.Game.Packets;
 using Maple2.Server.Game.Session;
+using System.Diagnostics;
```
Comment (Zintixx) [inline Maple2.Server.Game/PacketHandlers/ItemDismantleHandler.cs:10]: not being used. can be removed

File: Maple2.Server.Game/PacketHandlers/ItemDismantleHandler.cs [hunk 2]
```diff
@@ -173,10 +174,28 @@ private enum Command : byte {
...
+                        if (!item.Metadata.Limit.EnableBreak) {
+                            continue;
+                        }
+                        break;
+                    case "Outfit":
+                        if (item.GachaDismantleId == 0) {
+                            continue;
+                        }
+                        break;
+                }
```
Comment (Zintixx) [inline Maple2.Server.Game/PacketHandlers/ItemDismantleHandler.cs:197]: i'm not totally a fan of `ToString`-ing an enum. It allows for vulnerabilities if we ever change the enum names. You can still do this switch but just by the enum instead.

**Review by Zintixx — CHANGES_REQUESTED** [review]

**Review by Zintixx — APPROVED** [review]

**Comment by AngeloTadeucci** [comment]
just need to fix some extra whitespace and its good to go

**Pull request merged**